e2e/integration_tests/data_flow_test.go

```diff
@@ -242,6 +245,124 @@ func Test_EndpointLinksAreAvailable(t *testing.T) {
+	}
+	// expect this in db
+	//{
+	//	"id":3,
+	//	"chplProductNumber":"CHP-029177",
+	//	"edition":"2014",
+	//	"practiceType":"Inpatient",
+	//	"developer":"Intuitive Medical Documents",
+	//	"product":"Intuitive Medical Document",
+	//	"version":"2.0",
+	//	"certificationDate":1457049600000,
+	//	"certificationStatus":"Active",
+	//	"criteriaMet":"100☺101☺103☺106☺107☺108☺109☺114☺115☺116☺61☺62☺63☺64☺65☺66☺67☺68☺69☺70☺71☺72☺73☺74☺75☺76☺77☺81☺82☺83☺84☺86☺87☺88☺91☺92☺93☺94☺95☺96☺97☺98☺99"
+	//	}
+
+	var testHITP endpointmanager.HealthITProduct = endpointmanager.HealthITProduct{
+		Name:                  "Intuitive Medical Document",
+		Version:               "2.0",
+		Developer:             "Intuitive Medical Documents",
+		CertificationStatus:   "Active",
+		CertificationDate:     time.Date(2016, 3, 4, 0, 0, 0, 0, time.UTC),
+		CertificationEdition:  "2014",
+		CHPLID:                "CHP-029177",
+		CertificationCriteria: []string{"100", "101", "103", "106", "107", "108", "109", "114", "115", "116", "61", "62", "63", "64", "65", "66", "67", "68", "69", "70", "71", "72", "73", "74", "75", "76", "77", "81", "82", "83", "84", "86", "87", "88", "91", "92", "93", "94", "95", "96", "97", "98", "99"},
+	}
+
+	hitp, err := store.GetHealthITProductUsingNameAndVersion(ctx, "Intuitive Medical Document", "2.0")
+	th.Assert(t, err == nil, err)
+	th.Assert(t, hitp.CHPLID == testHITP.CHPLID, "CHPL ID is not what was expected")
+	th.Assert(t, hitp.CertificationEdition == testHITP.CertificationEdition, "Certification edition is not what was expected")
+	th.Assert(t, hitp.Developer == testHITP.Developer, "Developer is not what was expected")
+	th.Assert(t, hitp.CertificationDate.Equal(testHITP.CertificationDate), "Certification date is not what was expected")
+	th.Assert(t, hitp.CertificationStatus == testHITP.CertificationStatus, "Certification status is not what was expected")
+	th.Assert(t, reflect.DeepEqual(hitp.CertificationCriteria, testHITP.CertificationCriteria), "Certification criteria is not what was expected")
+}
+
+func Test_RetrieveCapabilityStatements(t *testing.T) {
+	var err error
+	qUser := viper.GetString("quser")
+	qPassword := viper.GetString("qpassword")
+	qHost := viper.GetString("qhost")
+	qPort := viper.GetString("qport")
+	qName := viper.GetString("qname")
+
+	hap := th.HostAndPort{Host: qHost, Port: qPort}
+	err = th.CheckResources(hap)
+	if err != nil {
+		panic(err)
+	}
+
+	var mq lanternmq.MessageQueue
+	var chID lanternmq.ChannelID
+	mq, chID, err = aq.ConnectToServerAndQueue(qUser, qPassword, qHost, qPort, qName)
+	defer mq.Close()
+	th.Assert(t, err == nil, err)
+	th.Assert(t, mq != nil, "expected message queue to be created")
+	th.Assert(t, chID != nil, "expected channel ID to be created")
+
+	ctx, cancel := context.WithCancel(context.Background())
+	go capabilityhandler.ReceiveCapabilityStatements(ctx, store, mq, chID, qName)
+	time.Sleep(30 * time.Second)
```

Comment:
It seems like context.WithDeadline or context.WithTimeout might make more sense here, but I am by no means a go expert. @edeyoung any thoughts?

Reply:
agree that `context.WithTimeout` is ideal.